### ISSUE TITLE
🐞 Fix build scripts 

### ISSUE THREAD

**Issue by ckunki** [exasol/python-toolbox#87]
### Checklist

- [x] I have reproduced the issue with at least the latest released version 0.5.0 of PTB.
- [x] I have added references to issues that sound similar in section _Related Issues_.

# Summary

1. References to workflows in files generated by PTB seem to be broken. In order to fix the CI build for repository [notebook-connector](https://github.com/exasol/notebook-connector/) we needed to replace  `exasol/python-toolbox/.github/workflows/build-and-publish.yml@0.5.0` by `./.github/workflows/build-and-publish.yml`

2. In file `.github/workflows/checks.yml` generated by PTB the indentation of the steps of `build-documentation-job` is wrong.

3. File `.github/workflows/checks.yml`, step "Check Version(s)" searches file `exasol/toolbox/version.py` while actually the github repo using the PTB should use a local path to its own version file, e.g. `version.py`

4. Missing test cases / files (e.g. `test/unit/test_smoke.py`) caused `pytest` to return exit code 5. 

5. missing configuration files and documentation caused build errors. I needed to create files `doc/conf.py` and `doc/index.rst`. Adding empty files fixed the build errors.

6. Missing folder artifacts" caused workflow `report.yml` to fail. 

7. In file `report.yml` step _Copy Artifacts into Root Folder_ failed to copy files that were missing.

8. Upload artifacts is missing in build script File  `.github/workflows/checks.yml`, after step `poetry run nox -s lint`

```
      - name: Upload Artifacts
        uses: actions/upload-artifact@v3
        with:
          name: .lint.txt
          path: .lint.txt
```

9. Upload artifacts is missing in build script File  `.github/workflows/checks.yml`, after step `poetry run nox -s coverage`

```
      - name: Upload Artifacts
        uses: actions/upload-artifact@v3
        with:
          name: .coverage
          path: .coverage
```

10. Some build steps in file `.github/workflows/checks.yml` have wrong name:

| step | current name | proposed name | 
|---|---|---|
| `poetry run nox -s lint` | Run Tests | Run Linter |
| `poetry run nox -s type-check` | Run Tests | Run Type check |
| `poetry run nox -s coverage` | Run Tests | Calculate Test Coverage |

------------------
Potential mitigations for missing files 
  * a) enhance PTB to generate default files with at least 1 test case 
  * b) enhance documentation to describe that folders must exist 

See repo https://github.com/exasol/notebook-connector/ for fixed version of the workflow files

## Reproducing the Issue

Reproducibility: always

### Steps to reproduce the behavior:
(See above)  

## Expected Behaviour
(see above) 

## Actual Behaviour
(see above) 

## Context
(see above) 

### System

Fedora, python 3, Github. 

**Desktop:**
 - OS: Fedora release 38 
 - Python 3.11.5
 - Package Version PTB 0.5.0

**Comment by ahsimb** [comment]
One more thing.

The pyproject.toml  must have a section similar to the one below.
```
[[tool.mypy.overrides]]
module = [
    "exasol.toolbox.nox.tasks",
    ...
]
ignore_missing_imports = true
```
Otherwise mypy complains about missing imports in the files imported by the noxfile.py.

**Comment by Nicoretti** [comment]
This issue does not appear to be a bug within the toolbox itself, but rather a configuration and project setup issue. Accordingly, most things will be addressed as part of [Issue #166](https://github.com/exasol/python-toolbox/issues/166) on our GitHub repository.

Side Note: version check have been changed in the latest `checks.yml`
